bigchaindb/crypto/fulfillments/ed25519_sha256.py

```diff
@@ -0,0 +1,266 @@
+import base58
+
+from bigchaindb.crypto.ed25519 import Ed25519VerifyingKey
+from bigchaindb.crypto.fulfillments.base_sha256 import BaseSha256Fulfillment
+from bigchaindb.crypto.buffer import Predictor
+
+
+class Ed25519Sha256Fulfillment(BaseSha256Fulfillment):
+
+    _bitmask = 0x08
+
+    def __init__(self):
+        self._message_prefix = None
+        self._max_dynamic_message_length = None
+        self._public_key = None
+        self._message = None
+        self._signature = None
+
+    @property
+    def message_prefix(self):
+        return self._message_prefix
+
+    @message_prefix.setter
+    def message_prefix(self, value):
+        """
+        Set the fixed message prefix.
+
+        The fixed prefix is the portion of the message that is determined when the
+        condition is first created.
+
+        Args:
+            value (Buffer): Static portion of the message
+        """
+        if not isinstance(value, bytes):
+            value = value.encode()
+        self._message_prefix = value
+
+    @property
+    def max_dynamic_message_length(self):
+        return self._max_dynamic_message_length
+
+    @max_dynamic_message_length.setter
+    def max_dynamic_message_length(self, value):
```

Comment:
Why do we need a `@property` and a `@<attrname>.setter`? What about having just the attribute in `self.max_dynamic_message_length`?